File: frontend/src/tasks/dto/TaskUpdateRequestDto.ts
import {TaskStatus} from "./TaskStatus.ts";
import {Importance} from "./Importance.ts";

export interface TaskUpdateRequestDto {
  readonly name: string;
  readonly description: string;
  readonly importance: Importance;
  readonly difficulty: number;
  readonly startDate: string;
  readonly deadline: string;
  readonly taskStatus: TaskStatus;
}

File: backend/src/main/java/net/dnadas/training_portal/dto/group/project/task/TaskUpdateRequestDto.java
package net.dnadas.training_portal.dto.group.project.task;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import net.dnadas.training_portal.model.group.project.task.Importance;
import net.dnadas.training_portal.model.group.project.task.TaskStatus;
import org.hibernate.validator.constraints.Length;

public record TaskUpdateRequestDto(
  @NotNull @Length(min = 1, max = 50) String name,
  @NotNull @Length(min = 1, max = 255) String description,
  @NotNull Importance importance,
  @NotNull @Min(1) @Max(5) Integer difficulty,
  @NotNull String startDate,
  @NotNull String deadline,
  @NotNull TaskStatus taskStatus) {
}
